collated/main/khooroko.md

```diff
@@ -0,0 +1,1424 @@
+        case 31:
+        case 32:
+        case 33:
+            return CLUSTER_POSTAL_DISTRICT_12;
+        case 34:
+        case 35:
+        case 36:
+        case 37:
+            return CLUSTER_POSTAL_DISTRICT_13;
+        case 38:
+        case 39:
+        case 40:
+        case 41:
+            return CLUSTER_POSTAL_DISTRICT_14;
+        case 42:
+        case 43:
+        case 44:
+        case 45:
+            return CLUSTER_POSTAL_DISTRICT_15;
+        case 46:
+        case 47:
+        case 48:
+            return CLUSTER_POSTAL_DISTRICT_16;
+        case 49:
+        case 50:
+        case 81:
+            return CLUSTER_POSTAL_DISTRICT_17;
+        case 51:
+        case 52:
+            return CLUSTER_POSTAL_DISTRICT_18;
+        case 53:
+        case 54:
+        case 55:
+        case 82:
+            return CLUSTER_POSTAL_DISTRICT_19;
+        case 56:
+        case 57:
+            return CLUSTER_POSTAL_DISTRICT_20;
+        case 58:
+        case 59:
+            return CLUSTER_POSTAL_DISTRICT_21;
+        case 60:
+        case 61:
+        case 62:
+        case 63:
+        case 64:
+            return CLUSTER_POSTAL_DISTRICT_22;
+        case 65:
+        case 66:
+        case 67:
+        case 68:
+            return CLUSTER_POSTAL_DISTRICT_23;
+        case 69:
+        case 70:
+        case 71:
+            return CLUSTER_POSTAL_DISTRICT_24;
+        case 72:
+        case 73:
+            return CLUSTER_POSTAL_DISTRICT_25;
+        case 77:
+        case 78:
+            return CLUSTER_POSTAL_DISTRICT_26;
+        case 75:
+        case 76:
+            return CLUSTER_POSTAL_DISTRICT_27;
+        case 79:
+        case 80:
+            return CLUSTER_POSTAL_DISTRICT_28;
+        default:
+            return CLUSTER_POSTAL_DISTRICT_UNKNOWN;
+        }
+    }
+
+}
+```
+###### \java\seedu\address\storage\XmlAddressBookStorage.java
+``` java
+    /**
+     * Creates a backup of the current data, if available.
+     */
+    @Override
+    public void backupAddressBook() {
+        try {
+            Optional<ReadOnlyAddressBook> addressBookOptional;
+            ReadOnlyAddressBook initialData;
+            addressBookOptional = readAddressBook();
+            if (!addressBookOptional.isPresent()) {
+                logger.info("No backup will be made as data file does not exist");
+            } else {
+                initialData = addressBookOptional.orElseGet(SampleDataUtil::getSampleAddressBook);
+                saveAddressBook(initialData, getBackupAddressBookFilePath());
+            }
+        } catch (DataConversionException e) {
+            logger.info("No backup will be made due to data file being in incorrect format");
+        } catch (IOException e) {
+            logger.info("No backup will be made due to problems while reading from file");
+        }
+    }
+
+    /**
+     * @return address book data, if available and readable, or backup address book data if it is not.
```

Comment:
The start of the header comment does not follow the convention (i.e. first sentence should be something like `Selects next line …` instead of `Select next line` or `This method is for selecting …`)